pkg/types/azure/machinepool.go

```diff
@@ -24,6 +24,12 @@ type MachinePool struct {
 	//
 	// +optional
 	OSDisk `json:"osDisk"`
+
+	// ultraSSDCapability defines if the instance should use Ultra SSD disks.
+	// The valid values are Enabled, Disabled.
+	//
+	// +optional
+	UltraSSDCapability string `json:"ultraSSDCapability,omitempty"`
```

Comment:
As this is essentially an enum, creating  a type for the valid values would make reading this a little easier. See Azure's `CloudEnvironment` for example: https://github.com/openshift/installer/blob/master/pkg/types/azure/platform.go#L89-L114

Reply:
I don't believe the readability benefit outweighs the cost in terms of binary size as well as additional CPU usage requirements to wrap a string in this way.